// captcha.gif@1.0.11/dist/index.js
'use strict';

var crypto = require('crypto');

var __defProp = Object.defineProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });

// src/lib/Colors.ts
var colors = [
  // Black 500 #000000
  `GIF89a\xC8\0F\0\x83\0\0















\xFF\xFF\xFF,\0\0\0\0\xC8\0F\0\0`,
  // Red 500 #F44336
  `GIF89a\xC8\0F\0\x83\0\0
\xF4C6
\xF4C6
\xF4C6
\xF4C6
\xF4C6
\xF4C6
\xF4C6
\xF4C6
\xF4C6
\xF4C6
\xF4C6
\xF4C6
\xF4C6
\xF4C6
\xF4C6
\xFF\xFF\xFF,\0\0\0\0\xC8\0F\0\0`,
  // Pink 500 #E91E63
  `GIF89a\xC8\0F\0\x83\0\0
\xE9c
\xE9c
\xE9c
\xE9c
\xE9c
\xE9c
\xE9c
\xE9c
\xE9c
\xE9c
\xE9c
\xE9c
\xE9c
\xE9c
\xE9c
\xFF\xFF\xFF,\0\0\0\0\xC8\0F\0\0`,
  // Purple 500 #9C27B0
  `GIF89a\xC8\0F\0\x83\0\0
\x9C'\xB0
\x9C'\xB0
\x9C'\xB0
\x9C'\xB0
\x9C'\xB0
\x9C'\xB0
\x9C'\xB0
\x9C'\xB0
\x9C'\xB0
\x9C'\xB0
\x9C'\xB0
\x9C'\xB0
\x9C'\xB0
\x9C'\xB0
\x9C'\xB0
\xFF\xFF\xFF,\0\0\0\0\xC8\0F\0\0`,
  // Deep Purple 500 #673AB7
  `GIF89a\xC8\0F\0\x83\0\0
g:\xB7
g:\xB7
g:\xB7
g:\xB7
g:\xB7
g:\xB7
g:\xB7
g:\xB7
g:\xB7
g:\xB7
g:\xB7
g:\xB7
g:\xB7
g:\xB7
g:\xB7
\xFF\xFF\xFF,\0\0\0\0\xC8\0F\0\0`,
  // Indigo 500 #3F51B5
  `GIF89a\xC8\0F\0\x83\0\0
?Q\xB5
?Q\xB5
?Q\xB5
?Q\xB5
?Q\xB5
?Q\xB5
?Q\xB5
?Q\xB5
?Q\xB5
?Q\xB5
?Q\xB5
?Q\xB5
?Q\xB5
?Q\xB5
?Q\xB5
\xFF\xFF\xFF,\0\0\0\0\xC8\0F\0\0`,
  // Blue 500 #2196F3
  `GIF89a\xC8\0F\0\x83\0\0
!\x96\xF3
!\x96\xF3
!\x96\xF3
!\x96\xF3
!\x96\xF3
!\x96\xF3
!\x96\xF3
!\x96\xF3
!\x96\xF3
!\x96\xF3
!\x96\xF3
!\x96\xF3
!\x96\xF3
!\x96\xF3
!\x96\xF3
\xFF\xFF\xFF,\0\0\0\0\xC8\0F\0\0`,
  // Light Blue 500 #03A9F4
  `GIF89a\xC8\0F\0\x83\0\0
\xA9\xF4
\xA9\xF4
\xA9\xF4
\xA9\xF4
\xA9\xF4
\xA9\xF4
\xA9\xF4
\xA9\xF4
\xA9\xF4
\xA9\xF4
\xA9\xF4
\xA9\xF4
\xA9\xF4
\xA9\xF4
\xA9\xF4
\xFF\xFF\xFF,\0\0\0\0\xC8\0F\0\0`,
  // Cyan 500 #00BCD4
  `GIF89a\xC8\0F\0\x83\0\0
\0\xBC\xD4
\0\xBC\xD4
\0\xBC\xD4
\0\xBC\xD4
\0\xBC\xD4
\0\xBC\xD4
\0\xBC\xD4
\0\xBC\xD4
\0\xBC\xD4
\0\xBC\xD4
\0\xBC\xD4
\0\xBC\xD4
\0\xBC\xD4
\0\xBC\xD4
\0\xBC\xD4
\xFF\xFF\xFF,\0\0\0\0\xC8\0F\0\0`,
  // Teal 500 #009688
  `GIF89a\xC8\0F\0\x83\0\0
\0\x96\x88
\0\x96\x88
\0\x96\x88
\0\x96\x88
\0\x96\x88
\0\x96\x88
\0\x96\x88
\0\x96\x88
\0\x96\x88
\0\x96\x88
\0\x96\x88
\0\x96\x88
\0\x96\x88
\0\x96\x88
\0\x96\x88
\xFF\xFF\xFF,\0\0\0\0\xC8\0F\0\0`,
  // Green 500 #4CAF50
  `GIF89a\xC8\0F\0\x83\0\0
L\xAFP
L\xAFP
L\xAFP
L\xAFP
L\xAFP
L\xAFP
L\xAFP
L\xAFP
L\xAFP
L\xAFP
L\xAFP
L\xAFP
L\xAFP
L\xAFP
L\xAFP
\xFF\xFF\xFF,\0\0\0\0\xC8\0F\0\0`,
  // Light Green 500 #8BC34A
  `GIF89a\xC8\0F\0\x83\0\0
\x8B\xC3J
\x8B\xC3J
\x8B\xC3J
\x8B\xC3J
\x8B\xC3J
\x8B\xC3J
\x8B\xC3J
\x8B\xC3J
\x8B\xC3J
\x8B\xC3J
\x8B\xC3J
\x8B\xC3J
\x8B\xC3J
\x8B\xC3J
\x8B\xC3J
\xFF\xFF\xFF,\0\0\0\0\xC8\0F\0\0`,
  // Lime 500 #CDDC39
  `GIF89a\xC8\0F\0\x83\0\0
\xCD\xDC9
\xCD\xDC9
\xCD\xDC9
\xCD\xDC9
\xCD\xDC9
\xCD\xDC9
\xCD\xDC9
\xCD\xDC9
\xCD\xDC9
\xCD\xDC9
\xCD\xDC9
\xCD\xDC9
\xCD\xDC9
\xCD\xDC9
\xCD\xDC9
\xFF\xFF\xFF,\0\0\0\0\xC8\0F\0\0`,
  // Yellow 500 #FFEB3B
  `GIF89a\xC8\0F\0\x83\0\0
\xFF\xEB;
\xFF\xEB;
\xFF\xEB;
\xFF\xEB;
\xFF\xEB;
\xFF\xEB;
\xFF\xEB;
\xFF\xEB;
\xFF\xEB;
\xFF\xEB;
\xFF\xEB;
\xFF\xEB;
\xFF\xEB;
\xFF\xEB;
\xFF\xEB;
\xFF\xFF\xFF,\0\0\0\0\xC8\0F\0\0`,
  // Amber 500 #FFC107
  `GIF89a\xC8\0F\0\x83\0\0
\xFF\xC1\x07
\xFF\xC1\x07
\xFF\xC1\x07
\xFF\xC1\x07
\xFF\xC1\x07
\xFF\xC1\x07
\xFF\xC1\x07
\xFF\xC1\x07
\xFF\xC1\x07
\xFF\xC1\x07
\xFF\xC1\x07
\xFF\xC1\x07
\xFF\xC1\x07
\xFF\xC1\x07
\xFF\xC1\x07
\xFF\xFF\xFF,\0\0\0\0\xC8\0F\0\0`,
  // Orange 500 #FF9800
  `GIF89a\xC8\0F\0\x83\0\0
\xFF\x98\0
\xFF\x98\0
\xFF\x98\0
\xFF\x98\0
\xFF\x98\0
\xFF\x98\0
\xFF\x98\0
\xFF\x98\0
\xFF\x98\0
\xFF\x98\0
\xFF\x98\0
\xFF\x98\0
\xFF\x98\0
\xFF\x98\0
\xFF\x98\0
\xFF\xFF\xFF,\0\0\0\0\xC8\0F\0\0`,
  // Deep Orange 500 #FF5722
  `GIF89a\xC8\0F\0\x83\0\0
\xFFW"
\xFFW"
\xFFW"
\xFFW"
\xFFW"
\xFFW"
\xFFW"
\xFFW"
\xFFW"
\xFFW"
\xFFW"
\xFFW"
\xFFW"
\xFFW"
\xFFW"
\xFF\xFF\xFF,\0\0\0\0\xC8\0F\0\0`,
  // Brown 500 #795548
  `GIF89a\xC8\0F\0\x83\0\0
yUH
yUH
yUH
yUH
yUH
yUH
yUH
yUH
yUH
yUH
yUH
yUH
yUH
yUH
yUH
\xFF\xFF\xFF,\0\0\0\0\xC8\0F\0\0`
];

// src/lib/Font.ts
var lt0 = [
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -4,
  11,
  7,
  5,
  3,
  1,
  0,
  0,
  0,
  1,
  3,
  7,
  13,
  -100,
  -2,
  11,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  -1,
  7,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  9,
  11,
  9,
  3,
  0,
  0,
  0,
  0,
  13,
  -100,
  9,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  -5,
  3,
  0,
  0,
  0,
  7,
  -100,
  5,
  0,
  0,
  0,
  0,
  0,
  1,
  13,
  -5,
  9,
  0,
  0,
  0,
  1,
  -100,
  7,
  0,
  0,
  0,
  0,
  1,
  13,
  -6,
  13,
  0,
  0,
  0,
  0,
  -100,
  -1,
  9,
  1,
  0,
  5,
  13,
  -8,
  0,
  0,
  0,
  0,
  13,
  -100,
  -14,
  0,
  0,
  0,
  0,
  11,
  -100,
  -14,
  0,
  0,
  0,
  0,
  11,
  -100,
  -14,
  0,
  0,
  0,
  0,
  11,
  -100,
  -12,
  13,
  5,
  0,
  0,
  0,
  0,
  11,
  -100,
  -8,
  13,
  9,
  5,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  13,
  7,
  3,
  1,
  0,
  0,
  0,
  0,
  1,
  1,
  0,
  0,
  0,
  0,
  11,
  -100,
  -2,
  13,
  5,
  0,
  0,
  0,
  0,
  0,
  5,
  9,
  13,
  -2,
  0,
  0,
  0,
  0,
  11,
  -100,
  -1,
  13,
  1,
  0,
  0,
  0,
  0,
  7,
  -6,
  0,
  0,
  0,
  0,
  11,
  -100,
  13,
  1,
  0,
  0,
  0,
  0,
  13,
  -7,
  0,
  0,
  0,
  0,
  11,
  -100,
  5,
  0,
  0,
  0,
  0,
  5,
  -8,
  0,
  0,
  0,
  0,
  11,
  -100,
  0,
  0,
  0,
  0,
  0,
  11,
  -8,
  0,
  0,
  0,
  0,
  11,
  -100,
  0,
  0,
  0,
  0,
  0,
  13,
  -7,
  13,
  0,
  0,
  0,
  0,
  11,
  -100,
  1,
  0,
  0,
  0,
  0,
  -7,
  9,
  0,
  0,
  0,
  0,
  0,
  9,
  -3,
  9,
  -100,
  5,
  0,
  0,
  0,
  0,
  3,
  13,
  -3,
  11,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  9,
  13,
  3,
  5,
  -100,
  13,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  1,
  0,
  0,
  0,
  1,
  11,
  9,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  13,
  -100,
  -1,
  11,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  -3,
  9,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  -2,
  13,
  7,
  3,
  0,
  0,
  0,
  3,
  7,
  13,
  -5,
  9,
  0,
  1,
  3,
  9,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -101
];
var lt1 = [
  -100,
  -100,
  -4,
  13,
  5,
  0,
  3,
  -100,
  -3,
  11,
  1,
  0,
  0,
  0,
  7,
  -100,
  -2,
  7,
  0,
  0,
  0,
  0,
  0,
  3,
  -100,
  13,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  -100,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  13,
  -100,
  13,
  3,
  0,
  0,
  0,
  0,
  0,
  1,
  -100,
  -2,
  5,
  0,
  0,
  0,
  0,
  5,
  -100,
  -3,
  0,
  0,
  0,
  0,
  9,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  9,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  3,
  13,
  9,
  5,
  3,
  1,
  0,
  0,
  1,
  3,
  5,
  9,
  -100,
  -3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  13,
  -100,
  -3,
  0,
  0,
  0,
  0,
  0,
  5,
  13,
  -2,
  13,
  11,
  9,
  5,
  0,
  0,
  0,
  0,
  1,
  13,
  -100,
  -3,
  0,
  0,
  0,
  0,
  1,
  -8,
  13,
  1,
  0,
  0,
  0,
  5,
  -100,
  -3,
  3,
  0,
  0,
  0,
  5,
  -9,
  13,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  9,
  -10,
  7,
  0,
  0,
  0,
  5,
  -100,
  -3,
  1,
  0,
  0,
  0,
  11,
  -10,
  13,
  0,
  0,
  0,
  0,
  -100,
  -3,
  3,
  0,
  0,
  0,
  11,
  -11,
  3,
  0,
  0,
  0,
  11,
  -100,
  -3,
  1,
  0,
  0,
  0,
  11,
  -11,
  7,
  0,
  0,
  0,
  7,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -11,
  9,
  0,
  0,
  0,
  3,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -11,
  11,
  0,
  0,
  0,
  1,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -11,
  11,
  0,
  0,
  0,
  1,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -11,
  11,
  0,
  0,
  0,
  0,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -11,
  9,
  0,
  0,
  0,
  0,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -11,
  7,
  0,
  0,
  0,
  3,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -11,
  3,
  0,
  0,
  0,
  7,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -11,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -10,
  9,
  0,
  0,
  0,
  3,
  -100,
  -3,
  0,
  0,
  0,
  0,
  9,
  -10,
  3,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  3,
  -9,
  11,
  0,
  0,
  0,
  5,
  -100,
  -2,
  13,
  0,
  0,
  0,
  0,
  0,
  9,
  -7,
  11,
  1,
  0,
  0,
  3,
  -100,
  -2,
  7,
  0,
  0,
  0,
  0,
  0,
  0,
  7,
  13,
  -2,
  13,
  9,
  3,
  0,
  0,
  0,
  3,
  13,
  -100,
  -2,
  13,
  0,
  0,
  5,
  13,
  11,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  7,
  -100,
  -3,
  9,
  11,
  -4,
  7,
  3,
  1,
  0,
  0,
  1,
  5,
  9,
  13,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -101
];
var lt2 = [
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -6,
  13,
  7,
  3,
  1,
  0,
  0,
  1,
  3,
  7,
  13,
  -100,
  -5,
  9,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  13,
  -100,
  -3,
  13,
  3,
  0,
  0,
  0,
  5,
  13,
  13,
  7,
  0,
  0,
  0,
  0,
  0,
  1,
  13,
  -100,
  -2,
  13,
  1,
  0,
  0,
  0,
  9,
  -4,
  7,
  0,
  0,
  0,
  0,
  0,
  1,
  -100,
  -1,
  13,
  1,
  0,
  0,
  0,
  9,
  -6,
  5,
  0,
  0,
  0,
  0,
  0,
  13,
  -100,
  -1,
  5,
  0,
  0,
  0,
  3,
  -8,
  1,
  0,
  0,
  0,
  3,
  -100,
  13,
  0,
  0,
  0,
  0,
  11,
  -8,
  13,
  3,
  0,
  3,
  -100,
  7,
  0,
  0,
  0,
  1,
  -100,
  5,
  0,
  0,
  0,
  5,
  -100,
  3,
  0,
  0,
  0,
  9,
  -100,
  1,
  0,
  0,
  0,
  11,
  -100,
  0,
  0,
  0,
  0,
  11,
  -100,
  0,
  0,
  0,
  0,
  11,
  -100,
  0,
  0,
  0,
  0,
  11,
  -100,
  0,
  0,
  0,
  0,
  9,
  -100,
  1,
  0,
  0,
  0,
  5,
  -100,
  5,
  0,
  0,
  0,
  0,
  13,
  -100,
  11,
  0,
  0,
  0,
  0,
  7,
  -100,
  -1,
  3,
  0,
  0,
  0,
  0,
  13,
  -100,
  -1,
  11,
  0,
  0,
  0,
  0,
  3,
  -12,
  9,
  -100,
  -2,
  7,
  0,
  0,
  0,
  0,
  3,
  13,
  -8,
  9,
  1,
  3,
  -100,
  -3,
  5,
  0,
  0,
  0,
  0,
  1,
  9,
  -5,
  9,
  3,
  0,
  0,
  11,
  -100,
  -4,
  5,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  1,
  1,
  0,
  0,
  0,
  0,
  11,
  -100,
  -5,
  9,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  13,
  -100,
  -7,
  11,
  7,
  3,
  1,
  0,
  1,
  3,
  7,
  11,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -101
];
var lt3 = [
  -100,
  -100,
  -100,
  -100,
  -18,
  11,
  3,
  0,
  -100,
  -16,
  13,
  3,
  0,
  0,
  0,
  -100,
  -14,
  9,
  3,
  0,
  0,
  0,
  0,
  0,
  -100,
  -13,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  -100,
  -13,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  -100,
  -13,
  9,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  -100,
  -15,
  13,
  5,
  0,
  0,
  0,
  0,
  -100,
  -17,
  1,
  0,
  0,
  0,
  -100,
  -17,
  3,
  0,
  0,
  0,
  -100,
  -17,
  1,
  0,
  0,
  0,
  -100,
  -17,
  0,
  0,
  0,
  0,
  -100,
  -16,
  13,
  0,
  0,
  0,
  0,
  -100,
  -16,
  13,
  0,
  0,
  0,
  0,
  -100,
  -6,
  11,
  5,
  3,
  1,
  0,
  0,
  1,
  5,
  9,
  13,
  13,
  0,
  0,
  0,
  0,
  -100,
  -4,
  13,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  0,
  0,
  0,
  0,
  -100,
  -3,
  13,
  1,
  0,
  0,
  0,
  0,
  5,
  11,
  -1,
  13,
  11,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  -100,
  -2,
  13,
  1,
  0,
  0,
  0,
  0,
  7,
  -6,
  11,
  0,
  0,
  0,
  0,
  0,
  -100,
  -2,
  3,
  0,
  0,
  0,
  0,
  9,
  -8,
  11,
  0,
  0,
  0,
  0,
  -100,
  -1,
  9,
  0,
  0,
  0,
  0,
  7,
  -9,
  11,
  0,
  0,
  0,
  0,
  -100,
  -1,
  1,
  0,
  0,
  0,
  0,
  13,
  -9,
  11,
  0,
  0,
  0,
  0,
  -100,
  11,
  0,
  0,
  0,
  0,
  3,
  -10,
  11,
  0,
  0,
  0,
  0,
  -100,
  5,
  0,
  0,
  0,
  0,
  13,
  -10,
  11,
  0,
  0,
  0,
  0,
  -100,
  3,
  0,
  0,
  0,
  3,
  -11,
  11,
  0,
  0,
  0,
  0,
  -100,
  0,
  0,
  0,
  0,
  7,
  -11,
  11,
  0,
  0,
  0,
  0,
  -100,
  0,
  0,
  0,
  0,
  11,
  -11,
  11,
  0,
  0,
  0,
  0,
  -100,
  0,
  0,
  0,
  0,
  11,
  -11,
  11,
  0,
  0,
  0,
  0,
  -100,
  0,
  0,
  0,
  0,
  11,
  -11,
  11,
  0,
  0,
  0,
  0,
  -100,
  1,
  0,
  0,
  0,
  9,
  -11,
  11,
  0,
  0,
  0,
  0,
  -100,
  3,
  0,
  0,
  0,
  7,
  -11,
  11,
  0,
  0,
  0,
  0,
  -100,
  7,
  0,
  0,
  0,
  3,
  -11,
  11,
  0,
  0,
  0,
  0,
  -100,
  13,
  0,
  0,
  0,
  0,
  13,
  -10,
  13,
  0,
  0,
  0,
  0,
  -100,
  -1,
  3,
  0,
  0,
  0,
  5,
  -11,
  0,
  0,
  0,
  0,
  -100,
  -1,
  13,
  0,
  0,
  0,
  0,
  11,
  -10,
  0,
  0,
  0,
  0,
  -100,
  -2,
  7,
  0,
  0,
  0,
  1,
  13,
  -8,
  13,
  0,
  0,
  0,
  0,
  13,
  -100,
  -3,
  5,
  0,
  0,
  0,
  1,
  11,
  -7,
  5,
  0,
  0,
  0,
  0,
  9,
  -100,
  -4,
  5,
  0,
  0,
  0,
  0,
  3,
  9,
  13,
  -1,
  13,
  11,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  -100,
  -5,
  9,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  13,
  -100,
  -7,
  9,
  3,
  1,
  0,
  0,
  3,
  7,
  13,
  -1,
  11,
  1,
  0,
  1,
  5,
  7,
  11,
  -100,
  -18,
  13,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -101
];
var lt4 = [
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -7,
  13,
  5,
  1,
  1,
  0,
  1,
  3,
  9,
  13,
  -100,
  -6,
  5,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  -4,
  11,
  1,
  0,
  0,
  3,
  11,
  -1,
  13,
  7,
  1,
  0,
  0,
  0,
  9,
  -100,
  -3,
  9,
  0,
  0,
  0,
  1,
  13,
  -5,
  3,
  0,
  0,
  1,
  -100,
  -2,
  9,
  0,
  0,
  0,
  0,
  13,
  -6,
  11,
  0,
  0,
  0,
  9,
  -100,
  -1,
  13,
  0,
  0,
  0,
  0,
  9,
  -8,
  0,
  0,
  0,
  1,
  -100,
  -1,
  5,
  0,
  0,
  0,
  0,
  13,
  -8,
  3,
  0,
  0,
  0,
  11,
  -100,
  -1,
  0,
  0,
  0,
  0,
  0,
  -8,
  13,
  1,
  0,
  0,
  0,
  5,
  -100,
  9,
  0,
  0,
  0,
  0,
  0,
  3,
  11,
  -4,
  13,
  11,
  1,
  0,
  0,
  0,
  0,
  1,
  -100,
  5,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  -100,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  -100,
  1,
  0,
  0,
  0,
  1,
  13,
  -100,
  0,
  0,
  0,
  0,
  11,
  -100,
  0,
  0,
  0,
  0,
  11,
  -100,
  1,
  0,
  0,
  0,
  9,
  -100,
  3,
  0,
  0,
  0,
  7,
  -100,
  7,
  0,
  0,
  0,
  5,
  -100,
  13,
  0,
  0,
  0,
  0,
  11,
  -100,
  -1,
  5,
  0,
  0,
  0,
  1,
  13,
  -100,
  -1,
  13,
  1,
  0,
  0,
  0,
  1,
  13,
  -10,
  13,
  11,
  -100,
  -2,
  11,
  0,
  0,
  0,
  0,
  1,
  13,
  -8,
  13,
  1,
  1,
  -100,
  -3,
  11,
  0,
  0,
  0,
  0,
  0,
  7,
  13,
  -4,
  11,
  7,
  0,
  0,
  9,
  -100,
  -4,
  11,
  1,
  0,
  0,
  0,
  0,
  0,
  1,
  1,
  1,
  0,
  0,
  0,
  0,
  7,
  -100,
  -5,
  13,
  5,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  9,
  -100,
  -7,
  13,
  7,
  3,
  0,
  0,
  0,
  1,
  5,
  9,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -101
];
var lt5 = [
  -100,
  -100,
  -100,
  -100,
  -9,
  13,
  9,
  3,
  1,
  0,
  1,
  5,
  13,
  -100,
  -8,
  7,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  -100,
  -7,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  -100,
  -6,
  5,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  -100,
  -5,
  13,
  0,
  0,
  3,
  13,
  9,
  1,
  0,
  0,
  0,
  0,
  11,
  -100,
  -5,
  9,
  0,
  0,
  13,
  -2,
  13,
  5,
  0,
  1,
  9,
  -100,
  -5,
  5,
  0,
  0,
  -100,
  -5,
  1,
  0,
  1,
  -100,
  -5,
  1,
  0,
  0,
  -100,
  -5,
  0,
  0,
  0,
  -100,
  -5,
  0,
  0,
  0,
  13,
  -100,
  -5,
  0,
  0,
  0,
  9,
  -100,
  -5,
  0,
  0,
  0,
  7,
  -100,
  -5,
  0,
  0,
  0,
  5,
  -100,
  -3,
  13,
  7,
  0,
  0,
  0,
  0,
  9,
  -1,
  13,
  11,
  13,
  -100,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  -1,
  11,
  13,
  -1,
  7,
  0,
  0,
  0,
  0,
  5,
  -1,
  13,
  11,
  13,
  -100,
  -5,
  0,
  0,
  0,
  0,
  11,
  -100,
  -5,
  0,
  0,
  0,
  0,
  11,
  -100,
  -5,
  0,
  0,
  0,
  0,
  11,
  -100,
  -5,
  0,
  0,
  0,
  0,
  11,
  -100,
  -5,
  0,
  0,
  0,
  0,
  11,
  -100,
  -5,
  0,
  0,
  0,
  0,
  11,
  -100,
  -5,
  0,
  0,
  0,
  0,
  11,
  -100,
  -5,
  0,
  0,
  0,
  0,
  11,
  -100,
  -5,
  0,
  0,
  0,
  0,
  11,
  -100,
  -5,
  0,
  0,
  0,
  0,
  11,
  -100,
  -5,
  0,
  0,
  0,
  0,
  11,
  -100,
  -5,
  0,
  0,
  0,
  0,
  11,
  -100,
  -5,
  0,
  0,
  0,
  0,
  11,
  -100,
  -5,
  0,
  0,
  0,
  0,
  11,
  -100,
  -5,
  0,
  0,
  0,
  0,
  11,
  -100,
  -5,
  0,
  0,
  0,
  0,
  7,
  -100,
  -3,
  13,
  5,
  0,
  0,
  0,
  0,
  0,
  7,
  13,
  -100,
  -1,
  5,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  11,
  -100,
  -1,
  7,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  3,
  7,
  13,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -101
];
var lt6 = [
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -7,
  11,
  7,
  3,
  1,
  0,
  0,
  1,
  3,
  7,
  13,
  -4,
  13,
  7,
  5,
  7,
  11,
  -100,
  -5,
  11,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  13,
  13,
  5,
  0,
  0,
  0,
  0,
  0,
  -100,
  -4,
  9,
  0,
  0,
  0,
  5,
  13,
  -1,
  13,
  9,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  -100,
  -3,
  11,
  0,
  0,
  0,
  5,
  -6,
  5,
  0,
  0,
  0,
  0,
  0,
  11,
  13,
  9,
  3,
  0,
  9,
  -100,
  -2,
  13,
  0,
  0,
  0,
  0,
  13,
  -7,
  1,
  0,
  0,
  0,
  0,
  13,
  -100,
  -2,
  1,
  0,
  0,
  0,
  7,
  -8,
  7,
  0,
  0,
  0,
  0,
  7,
  -100,
  -2,
  0,
  0,
  0,
  0,
  9,
  -8,
  11,
  0,
  0,
  0,
  0,
  3,
  -100,
  -2,
  0,
  0,
  0,
  0,
  11,
  -9,
  0,
  0,
  0,
  0,
  1,
  -100,
  -2,
  0,
  0,
  0,
  0,
  11,
  -9,
  0,
  0,
  0,
  0,
  0,
  -100,
  -2,
  0,
  0,
  0,
  0,
  9,
  -8,
  13,
  0,
  0,
  0,
  0,
  1,
  -100,
  -2,
  3,
  0,
  0,
  0,
  5,
  -8,
  9,
  0,
  0,
  0,
  0,
  7,
  -100,
  -2,
  13,
  1,
  0,
  0,
  0,
  13,
  -7,
  5,
  0,
  0,
  0,
  0,
  13,
  -100,
  -3,
  11,
  0,
  0,
  0,
  3,
  13,
  -5,
  7,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  7,
  0,
  0,
  0,
  1,
  9,
  13,
  -1,
  13,
  7,
  0,
  0,
  0,
  1,
  11,
  -100,
  -4,
  13,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  13,
  -100,
  -4,
  13,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  3,
  5,
  9,
  -100,
  -2,
  13,
  5,
  0,
  11,
  -100,
  -1,
  13,
  1,
  0,
  0,
  -100,
  -1,
  3,
  0,
  0,
  0,
  1,
  7,
  13,
  -100,
  -1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  3,
  7,
  9,
  11,
  13,
  -100,
  -1,
  5,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  3,
  5,
  11,
  -100,
  -2,
  11,
  7,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  11,
  -100,
  -4,
  13,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  -3,
  9,
  1,
  0,
  3,
  9,
  13,
  13,
  11,
  7,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  13,
  -100,
  -1,
  7,
  1,
  0,
  0,
  9,
  -8,
  11,
  7,
  3,
  0,
  0,
  0,
  0,
  0,
  5,
  -100,
  9,
  0,
  0,
  0,
  3,
  -13,
  11,
  1,
  0,
  0,
  1,
  -100,
  1,
  0,
  0,
  0,
  9,
  -15,
  0,
  0,
  0,
  -100,
  0,
  0,
  0,
  0,
  9,
  -14,
  13,
  0,
  0,
  1,
  -100,
  3,
  0,
  0,
  0,
  1,
  -14,
  7,
  0,
  0,
  5,
  -100,
  11,
  0,
  0,
  0,
  0,
  5,
  -12,
  11,
  0,
  0,
  0,
  13,
  -100,
  -1,
  11,
  1,
  0,
  0,
  0,
  1,
  7,
  11,
  13,
  -1,
  13,
  13,
  11,
  9,
  5,
  3,
  1,
  0,
  0,
  3,
  13,
  -100,
  -3,
  9,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  9,
  -100,
  -6,
  9,
  7,
  3,
  1,
  1,
  0,
  0,
  1,
  1,
  3,
  7,
  11,
  -100,
  -101
];
var lt7 = [
  -100,
  -100,
  -100,
  -100,
  -4,
  13,
  7,
  1,
  9,
  -100,
  -2,
  13,
  7,
  0,
  0,
  0,
  3,
  -100,
  -1,
  7,
  0,
  0,
  0,
  0,
  0,
  5,
  -100,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  7,
  -100,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  7,
  -100,
  13,
  7,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  -2,
  5,
  0,
  0,
  0,
  0,
  9,
  -100,
  -2,
  11,
  0,
  0,
  0,
  0,
  9,
  -100,
  -3,
  0,
  0,
  0,
  0,
  9,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -5,
  11,
  5,
  1,
  0,
  1,
  7,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -3,
  13,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -2,
  7,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  11,
  3,
  0,
  5,
  11,
  -1,
  13,
  9,
  1,
  0,
  0,
  0,
  3,
  -100,
  -3,
  0,
  0,
  0,
  0,
  3,
  0,
  1,
  11,
  -6,
  1,
  0,
  0,
  0,
  13,
  -100,
  -3,
  0,
  0,
  0,
  0,
  0,
  5,
  -8,
  11,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -10,
  0,
  0,
  0,
  7,
  -100,
  -3,
  0,
  0,
  0,
  0,
  13,
  -10,
  0,
  0,
  0,
  7,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -10,
  0,
  0,
  0,
  7,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -10,
  0,
  0,
  0,
  7,
  -100,
  -3,
  0,
  0,
  0,
  0,
  13,
  -10,
  0,
  0,
  0,
  9,
  -100,
  -3,
  0,
  0,
  0,
  0,
  13,
  -10,
  0,
  0,
  0,
  9,
  -100,
  -3,
  0,
  0,
  0,
  0,
  -11,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  1,
  -11,
  0,
  0,
  0,
  13,
  -100,
  -3,
  0,
  0,
  0,
  1,
  -11,
  0,
  0,
  0,
  13,
  -100,
  -3,
  0,
  0,
  0,
  3,
  -11,
  0,
  0,
  0,
  -100,
  -3,
  0,
  0,
  0,
  3,
  -11,
  0,
  0,
  0,
  -100,
  -3,
  0,
  0,
  0,
  3,
  -11,
  0,
  0,
  0,
  -100,
  -3,
  0,
  0,
  0,
  3,
  -10,
  13,
  0,
  0,
  0,
  13,
  -100,
  -3,
  0,
  0,
  0,
  1,
  -10,
  13,
  0,
  0,
  0,
  11,
  -100,
  -2,
  9,
  0,
  0,
  0,
  0,
  -10,
  11,
  0,
  0,
  0,
  5,
  -100,
  -1,
  7,
  0,
  0,
  0,
  0,
  0,
  5,
  13,
  -7,
  13,
  3,
  0,
  0,
  0,
  0,
  7,
  -100,
  5,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  -5,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  3,
  11,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -101
];
var lt8 = [
  -100,
  -100,
  -100,
  -100,
  -3,
  5,
  1,
  0,
  3,
  11,
  -100,
  -2,
  7,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  -2,
  1,
  0,
  0,
  0,
  0,
  0,
  5,
  -100,
  -2,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  -100,
  -2,
  5,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  7,
  0,
  0,
  3,
  9,
  -100,
  -4,
  13,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -4,
  11,
  3,
  0,
  9,
  -100,
  -2,
  9,
  3,
  0,
  0,
  0,
  9,
  -100,
  11,
  1,
  0,
  0,
  0,
  0,
  0,
  7,
  -100,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  7,
  -100,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  -1,
  11,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  -2,
  7,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  13,
  -100,
  -3,
  1,
  0,
  0,
  0,
  -100,
  -3,
  1,
  0,
  0,
  0,
  13,
  -100,
  -3,
  1,
  0,
  0,
  0,
  13,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -2,
  13,
  0,
  0,
  0,
  0,
  7,
  -100,
  -2,
  13,
  0,
  0,
  0,
  0,
  1,
  -100,
  -2,
  7,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  7,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  -100,
  9,
  5,
  1,
  1,
  0,
  0,
  1,
  1,
  3,
  5,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -101
];
var lt9 = [
  -100,
  -100,
  -100,
  -100,
  -6,
  5,
  0,
  1,
  5,
  13,
  -100,
  -5,
  5,
  0,
  0,
  0,
  0,
  1,
  -100,
  -5,
  1,
  0,
  0,
  0,
  0,
  0,
  13,
  -100,
  -5,
  0,
  0,
  0,
  0,
  0,
  0,
  -100,
  -5,
  5,
  0,
  0,
  0,
  0,
  3,
  -100,
  -6,
  7,
  1,
  0,
  5,
  13,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -6,
  13,
  7,
  3,
  0,
  13,
  -100,
  -5,
  7,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  7,
  1,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  -2,
  13,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  11,
  3,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  -5,
  9,
  0,
  0,
  0,
  0,
  11,
  -100,
  -6,
  1,
  0,
  0,
  0,
  11,
  -100,
  -6,
  3,
  0,
  0,
  0,
  11,
  -100,
  -6,
  5,
  0,
  0,
  0,
  11,
  -100,
  -6,
  5,
  0,
  0,
  0,
  11,
  -100,
  -6,
  5,
  0,
  0,
  0,
  11,
  -100,
  -6,
  5,
  0,
  0,
  0,
  11,
  -100,
  -6,
  3,
  0,
  0,
  0,
  11,
  -100,
  -6,
  3,
  0,
  0,
  0,
  11,
  -100,
  -6,
  1,
  0,
  0,
  0,
  11,
  -100,
  -6,
  1,
  0,
  0,
  0,
  13,
  -100,
  -6,
  1,
  0,
  0,
  0,
  13,
  -100,
  -6,
  1,
  0,
  0,
  0,
  13,
  -100,
  -6,
  1,
  0,
  0,
  0,
  13,
  -100,
  -6,
  1,
  0,
  0,
  0,
  13,
  -100,
  -6,
  1,
  0,
  0,
  0,
  -100,
  -6,
  1,
  0,
  0,
  0,
  -100,
  -6,
  1,
  0,
  0,
  1,
  -100,
  -6,
  1,
  0,
  0,
  3,
  -100,
  -6,
  0,
  0,
  0,
  5,
  -100,
  -6,
  0,
  0,
  0,
  7,
  -100,
  -6,
  0,
  0,
  0,
  11,
  -100,
  11,
  0,
  0,
  0,
  5,
  5,
  0,
  0,
  5,
  -100,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  -100,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  13,
  -100,
  1,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  13,
  3,
  0,
  0,
  3,
  11,
  -100,
  -100,
  -101
];
var lt10 = [
  -100,
  -100,
  -100,
  -100,
  -6,
  9,
  1,
  13,
  -100,
  -3,
  13,
  5,
  0,
  0,
  0,
  11,
  -100,
  -1,
  13,
  5,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  13,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  11,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  -2,
  13,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  9,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -7,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -7,
  5,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  7,
  13,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -8,
  9,
  0,
  0,
  0,
  1,
  9,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -7,
  13,
  1,
  0,
  0,
  7,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -7,
  5,
  0,
  3,
  13,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -6,
  5,
  0,
  5,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -5,
  5,
  1,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -3,
  11,
  3,
  1,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -1,
  11,
  3,
  0,
  0,
  1,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  9,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  0,
  0,
  0,
  0,
  3,
  -100,
  -4,
  0,
  0,
  0,
  0,
  0,
  9,
  -1,
  11,
  1,
  0,
  0,
  0,
  9,
  -100,
  -4,
  0,
  0,
  0,
  0,
  5,
  -4,
  3,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -4,
  13,
  1,
  0,
  0,
  0,
  13,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -5,
  11,
  0,
  0,
  0,
  1,
  13,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -6,
  3,
  0,
  0,
  0,
  3,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -6,
  11,
  0,
  0,
  0,
  0,
  7,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -7,
  3,
  0,
  0,
  0,
  0,
  7,
  -100,
  -3,
  13,
  0,
  0,
  0,
  0,
  11,
  -7,
  13,
  0,
  0,
  0,
  0,
  0,
  7,
  -100,
  -2,
  13,
  3,
  0,
  0,
  0,
  0,
  7,
  -7,
  9,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  11,
  -100,
  -1,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  3,
  -4,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  -100,
  -1,
  7,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  7,
  -4,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  1,
  5,
  9,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -101
];
var lt11 = [
  -100,
  -100,
  -100,
  -6,
  11,
  7,
  11,
  -100,
  -4,
  11,
  3,
  0,
  0,
  5,
  -100,
  -3,
  5,
  0,
  0,
  0,
  0,
  5,
  -100,
  -1,
  11,
  1,
  0,
  0,
  0,
  0,
  0,
  7,
  -100,
  7,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  7,
  -100,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  7,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  -2,
  9,
  1,
  0,
  0,
  0,
  0,
  9,
  -100,
  -3,
  11,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  7,
  -100,
  -3,
  5,
  0,
  0,
  0,
  0,
  0,
  13,
  -100,
  -1,
  9,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  9,
  -100,
  -1,
  5,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  11,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -101
];
var lt12 = [
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -5,
  11,
  5,
  0,
  -21,
  7,
  1,
  0,
  3,
  9,
  -100,
  -4,
  7,
  0,
  0,
  0,
  13,
  -4,
  11,
  5,
  1,
  1,
  0,
  1,
  7,
  -7,
  11,
  3,
  0,
  0,
  0,
  0,
  0,
  5,
  -100,
  -2,
  11,
  1,
  0,
  0,
  0,
  0,
  11,
  -2,
  9,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  9,
  -4,
  13,
  5,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  7,
  -100,
  13,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  7,
  13,
  3,
  0,
  0,
  0,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  7,
  -1,
  13,
  7,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  0,
  0,
  5,
  11,
  -3,
  11,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  9,
  13,
  13,
  -3,
  7,
  0,
  0,
  0,
  0,
  5,
  -100,
  13,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  7,
  13,
  -6,
  3,
  0,
  0,
  0,
  0,
  0,
  9,
  -8,
  5,
  0,
  0,
  0,
  1,
  -100,
  -2,
  11,
  0,
  0,
  0,
  0,
  0,
  3,
  -8,
  9,
  0,
  0,
  0,
  0,
  7,
  -9,
  11,
  0,
  0,
  0,
  0,
  -100,
  -3,
  3,
  0,
  0,
  0,
  0,
  7,
  -8,
  13,
  0,
  0,
  0,
  0,
  -10,
  11,
  0,
  0,
  0,
  0,
  -100,
  -3,
  7,
  0,
  0,
  0,
  0,
  11,
  -9,
  0,
  0,
  0,
  0,
  -10,
  11,
  0,
  0,
  0,
  1,
  -100,
  -3,
  11,
  0,
  0,
  0,
  0,
  11,
  -9,
  0,
  0,
  0,
  0,
  -10,
  11,
  0,
  0,
  0,
  3,
  -100,
  -3,
  13,
  0,
  0,
  0,
  0,
  11,
  -8,
  13,
  0,
  0,
  0,
  0,
  -10,
  11,
  0,
  0,
  0,
  5,
  -100,
  -3,
  13,
  0,
  0,
  0,
  0,
  11,
  -8,
  11,
  0,
  0,
  0,
  0,
  -10,
  11,
  0,
  0,
  0,
  5,
  -100,
  -4,
  0,
  0,
  0,
  0,
  13,
  -8,
  9,
  0,
  0,
  0,
  0,
  -10,
  11,
  0,
  0,
  0,
  5,
  -100,
  -4,
  0,
  0,
  0,
  0,
  13,
  -8,
  11,
  0,
  0,
  0,
  0,
  -10,
  11,
  0,
  0,
  0,
  3,
  -100,
  -4,
  0,
  0,
  0,
  0,
  13,
  -8,
  11,
  0,
  0,
  0,
  0,
  -10,
  11,
  0,
  0,
  0,
  0,
  -100,
  -3,
  13,
  0,
  0,
  0,
  1,
  -9,
  13,
  0,
  0,
  0,
  0,
  -10,
  11,
  0,
  0,
  0,
  0,
  -100,
  -3,
  11,
  0,
  0,
  0,
  13,
  -10,
  0,
  0,
  0,
  0,
  -10,
  11,
  0,
  0,
  0,
  0,
  -100,
  -4,
  0,
  0,
  0,
  11,
  -10,
  0,
  0,
  0,
  0,
  -10,
  11,
  0,
  0,
  0,
  0,
  -100,
  -3,
  13,
  0,
  0,
  0,
  3,
  -9,
  13,
  0,
  0,
  0,
  0,
  13,
  -9,
  9,
  0,
  0,
  0,
  0,
  -100,
  -3,
  7,
  0,
  0,
  0,
  0,
  13,
  -8,
  11,
  0,
  0,
  0,
  0,
  11,
  -9,
  7,
  0,
  0,
  0,
  0,
  13,
  -100,
  -3,
  3,
  0,
  0,
  0,
  0,
  9,
  -8,
  5,
  0,
  0,
  0,
  0,
  5,
  -9,
  5,
  0,
  0,
  0,
  0,
  7,
  -100,
  -1,
  13,
  3,
  0,
  0,
  0,
  0,
  0,
  1,
  11,
  -6,
  9,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  -7,
  11,
  1,
  0,
  0,
  0,
  0,
  1,
  13,
  -100,
  11,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  -4,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  -5,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  13,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  9,
  -4,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  5,
  -5,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  7,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -101
];
var lt13 = [
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -3,
  13,
  7,
  3,
  0,
  13,
  -5,
  13,
  7,
  1,
  0,
  0,
  3,
  7,
  -100,
  -2,
  5,
  0,
  0,
  0,
  0,
  -4,
  13,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  7,
  -100,
  7,
  1,
  0,
  0,
  0,
  0,
  0,
  -3,
  7,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  13,
  -100,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  13,
  9,
  1,
  0,
  5,
  11,
  13,
  -1,
  11,
  3,
  0,
  0,
  0,
  0,
  9,
  -100,
  13,
  3,
  0,
  0,
  0,
  0,
  0,
  1,
  1,
  7,
  -7,
  3,
  0,
  0,
  0,
  3,
  -100,
  -2,
  1,
  0,
  0,
  0,
  0,
  7,
  -9,
  13,
  0,
  0,
  0,
  1,
  -100,
  -2,
  7,
  0,
  0,
  0,
  0,
  -11,
  1,
  0,
  0,
  0,
  -100,
  -2,
  9,
  0,
  0,
  0,
  0,
  -11,
  1,
  0,
  0,
  0,
  13,
  -100,
  -2,
  11,
  0,
  0,
  0,
  0,
  -11,
  3,
  0,
  0,
  0,
  13,
  -100,
  -2,
  11,
  0,
  0,
  0,
  0,
  -11,
  5,
  0,
  0,
  0,
  11,
  -100,
  -2,
  11,
  0,
  0,
  0,
  0,
  -11,
  3,
  0,
  0,
  0,
  11,
  -100,
  -2,
  11,
  0,
  0,
  0,
  0,
  -11,
  1,
  0,
  0,
  0,
  11,
  -100,
  -2,
  11,
  0,
  0,
  0,
  0,
  -11,
  0,
  0,
  0,
  0,
  11,
  -100,
  -2,
  11,
  0,
  0,
  0,
  0,
  -11,
  0,
  0,
  0,
  0,
  11,
  -100,
  -2,
  11,
  0,
  0,
  0,
  0,
  -11,
  1,
  0,
  0,
  0,
  11,
  -100,
  -2,
  11,
  0,
  0,
  0,
  0,
  -11,
  3,
  0,
  0,
  0,
  11,
  -100,
  -2,
  11,
  0,
  0,
  0,
  0,
  -11,
  3,
  0,
  0,
  0,
  11,
  -100,
  -2,
  11,
  0,
  0,
  0,
  0,
  13,
  -10,
  5,
  0,
  0,
  0,
  11,
  -100,
  -2,
  11,
  0,
  0,
  0,
  0,
  13,
  -10,
  5,
  0,
  0,
  0,
  11,
  -100,
  -2,
  11,
  0,
  0,
  0,
  0,
  9,
  -10,
  5,
  0,
  0,
  0,
  9,
  -100,
  -2,
  7,
  0,
  0,
  0,
  0,
  5,
  -10,
  3,
  0,
  0,
  0,
  7,
  -100,
  -1,
  13,
  1,
  0,
  0,
  0,
  0,
  1,
  11,
  -7,
  11,
  5,
  0,
  0,
  0,
  0,
  0,
  3,
  9,
  -100,
  -1,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  -4,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  -100,
  -1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  -4,
  5,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -101
];
var lt14 = [
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -7,
  13,
  7,
  3,
  1,
  0,
  0,
  1,
  5,
  9,
  -100,
  -6,
  7,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  7,
  -100,
  -4,
  11,
  1,
  0,
  0,
  1,
  9,
  13,
  -1,
  13,
  11,
  5,
  0,
  0,
  0,
  3,
  13,
  -100,
  -3,
  11,
  0,
  0,
  0,
  3,
  13,
  -6,
  13,
  1,
  0,
  0,
  0,
  13,
  -100,
  -2,
  11,
  0,
  0,
  0,
  1,
  13,
  -8,
  13,
  0,
  0,
  0,
  5,
  -100,
  -2,
  0,
  0,
  0,
  0,
  9,
  -10,
  5,
  0,
  0,
  0,
  11,
  -100,
  -1,
  7,
  0,
  0,
  0,
  1,
  -11,
  13,
  0,
  0,
  0,
  1,
  -100,
  -1,
  1,
  0,
  0,
  0,
  5,
  -12,
  1,
  0,
  0,
  0,
  9,
  -100,
  11,
  0,
  0,
  0,
  0,
  11,
  -12,
  5,
  0,
  0,
  0,
  3,
  -100,
  7,
  0,
  0,
  0,
  0,
  13,
  -12,
  9,
  0,
  0,
  0,
  0,
  -100,
  3,
  0,
  0,
  0,
  0,
  13,
  -12,
  11,
  0,
  0,
  0,
  0,
  -100,
  1,
  0,
  0,
  0,
  0,
  -13,
  11,
  0,
  0,
  0,
  1,
  -100,
  0,
  0,
  0,
  0,
  0,
  -13,
  11,
  0,
  0,
  0,
  1,
  -100,
  0,
  0,
  0,
  0,
  0,
  13,
  -12,
  11,
  0,
  0,
  0,
  0,
  -100,
  1,
  0,
  0,
  0,
  0,
  13,
  -12,
  9,
  0,
  0,
  0,
  1,
  -100,
  5,
  0,
  0,
  0,
  0,
  11,
  -12,
  5,
  0,
  0,
  0,
  3,
  -100,
  9,
  0,
  0,
  0,
  0,
  9,
  -12,
  3,
  0,
  0,
  0,
  7,
  -100,
  -1,
  1,
  0,
  0,
  0,
  5,
  -12,
  0,
  0,
  0,
  0,
  11,
  -100,
  -1,
  9,
  0,
  0,
  0,
  3,
  -11,
  13,
  0,
  0,
  0,
  3,
  -100,
  -2,
  3,
  0,
  0,
  0,
  13,
  -10,
  7,
  0,
  0,
  0,
  11,
  -100,
  -2,
  13,
  1,
  0,
  0,
  3,
  -9,
  13,
  0,
  0,
  0,
  7,
  -100,
  -3,
  13,
  1,
  0,
  0,
  3,
  13,
  -6,
  13,
  1,
  0,
  0,
  5,
  -100,
  -4,
  13,
  1,
  0,
  0,
  1,
  7,
  13,
  -3,
  9,
  0,
  0,
  0,
  7,
  -100,
  -6,
  7,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  11,
  -100,
  -7,
  13,
  7,
  3,
  0,
  0,
  0,
  0,
  3,
  11,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -101
];
var lt15 = [
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -6,
  11,
  7,
  -100,
  -4,
  11,
  3,
  0,
  0,
  7,
  11,
  5,
  3,
  0,
  0,
  0,
  1,
  5,
  9,
  -100,
  -2,
  11,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  -100,
  13,
  5,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  7,
  11,
  13,
  -1,
  13,
  3,
  0,
  0,
  0,
  5,
  -100,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  -7,
  7,
  0,
  0,
  0,
  7,
  -100,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  11,
  -8,
  7,
  0,
  0,
  0,
  13,
  -100,
  -1,
  11,
  5,
  0,
  0,
  0,
  0,
  0,
  11,
  -9,
  1,
  0,
  0,
  3,
  -100,
  -3,
  7,
  0,
  0,
  0,
  0,
  11,
  -9,
  5,
  0,
  0,
  0,
  11,
  -100,
  -3,
  9,
  0,
  0,
  0,
  0,
  11,
  -9,
  9,
  0,
  0,
  0,
  5,
  -100,
  -3,
  11,
  0,
  0,
  0,
  0,
  11,
  -9,
  11,
  0,
  0,
  0,
  0,
  13,
  -100,
  -3,
  11,
  0,
  0,
  0,
  0,
  11,
  -9,
  13,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  13,
  0,
  0,
  0,
  0,
  11,
  -10,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  13,
  0,
  0,
  0,
  0,
  11,
  -10,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  13,
  0,
  0,
  0,
  0,
  11,
  -10,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -10,
  0,
  0,
  0,
  0,
  13,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -10,
  0,
  0,
  0,
  0,
  13,
  -100,
  -4,
  0,
  0,
  0,
  0,
  13,
  -10,
  0,
  0,
  0,
  0,
  -100,
  -4,
  0,
  0,
  0,
  9,
  -11,
  0,
  0,
  0,
  3,
  -100,
  -4,
  0,
  0,
  0,
  3,
  -11,
  0,
  0,
  0,
  9,
  -100,
  -4,
  0,
  0,
  0,
  0,
  13,
  -10,
  0,
  0,
  0,
  13,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -9,
  5,
  0,
  0,
  0,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -8,
  7,
  0,
  0,
  0,
  7,
  -100,
  -4,
  0,
  0,
  0,
  0,
  5,
  -7,
  9,
  0,
  0,
  0,
  3,
  -100,
  -4,
  0,
  0,
  0,
  0,
  0,
  3,
  11,
  13,
  -3,
  9,
  0,
  0,
  0,
  1,
  -100,
  -4,
  0,
  0,
  0,
  0,
  5,
  7,
  0,
  0,
  0,
  1,
  1,
  0,
  0,
  0,
  1,
  13,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -2,
  9,
  5,
  1,
  0,
  0,
  3,
  7,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  7,
  -100,
  -4,
  0,
  0,
  0,
  0,
  3,
  -100,
  -3,
  7,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  -1,
  5,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  -100,
  -1,
  3,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  -100,
  -101
];
var lt16 = [
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -19,
  7,
  11,
  -100,
  -6,
  11,
  5,
  1,
  0,
  0,
  1,
  5,
  11,
  -4,
  5,
  0,
  3,
  -100,
  -4,
  13,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  11,
  13,
  5,
  0,
  0,
  1,
  -100,
  -3,
  13,
  1,
  0,
  0,
  1,
  5,
  11,
  -3,
  11,
  3,
  0,
  0,
  0,
  0,
  0,
  3,
  -100,
  -2,
  13,
  1,
  0,
  0,
  3,
  -8,
  3,
  0,
  0,
  0,
  0,
  5,
  -100,
  -1,
  13,
  1,
  0,
  0,
  0,
  13,
  -8,
  7,
  0,
  0,
  0,
  0,
  5,
  -100,
  -1,
  3,
  0,
  0,
  0,
  5,
  -9,
  11,
  0,
  0,
  0,
  0,
  7,
  -100,
  9,
  0,
  0,
  0,
  0,
  11,
  -9,
  13,
  0,
  0,
  0,
  0,
  9,
  -100,
  3,
  0,
  0,
  0,
  0,
  -10,
  13,
  0,
  0,
  0,
  0,
  9,
  -100,
  1,
  0,
  0,
  0,
  1,
  -11,
  0,
  0,
  0,
  0,
  9,
  -100,
  0,
  0,
  0,
  0,
  3,
  -11,
  0,
  0,
  0,
  0,
  11,
  -100,
  0,
  0,
  0,
  0,
  5,
  -11,
  0,
  0,
  0,
  0,
  11,
  -100,
  0,
  0,
  0,
  0,
  9,
  -11,
  0,
  0,
  0,
  0,
  11,
  -100,
  0,
  0,
  0,
  0,
  11,
  -11,
  0,
  0,
  0,
  0,
  11,
  -100,
  0,
  0,
  0,
  0,
  11,
  -11,
  0,
  0,
  0,
  0,
  11,
  -100,
  1,
  0,
  0,
  0,
  11,
  -11,
  0,
  0,
  0,
  0,
  11,
  -100,
  3,
  0,
  0,
  0,
  9,
  -11,
  0,
  0,
  0,
  0,
  11,
  -100,
  7,
  0,
  0,
  0,
  5,
  -11,
  0,
  0,
  0,
  0,
  11,
  -100,
  11,
  0,
  0,
  0,
  0,
  -11,
  0,
  0,
  0,
  0,
  11,
  -100,
  -1,
  1,
  0,
  0,
  0,
  7,
  -9,
  11,
  0,
  0,
  0,
  0,
  11,
  -100,
  -1,
  9,
  0,
  0,
  0,
  0,
  13,
  -8,
  9,
  0,
  0,
  0,
  0,
  11,
  -100,
  -2,
  3,
  0,
  0,
  0,
  1,
  13,
  -7,
  5,
  0,
  0,
  0,
  0,
  11,
  -100,
  -2,
  13,
  3,
  0,
  0,
  0,
  1,
  9,
  13,
  -3,
  13,
  7,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  9,
  1,
  0,
  0,
  0,
  0,
  0,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  -6,
  11,
  7,
  3,
  1,
  0,
  0,
  5,
  9,
  -1,
  13,
  0,
  0,
  0,
  0,
  11,
  -100,
  -16,
  0,
  0,
  0,
  0,
  13,
  -100,
  -16,
  0,
  0,
  0,
  0,
  13,
  -100,
  -16,
  0,
  0,
  0,
  0,
  13,
  -100,
  -16,
  0,
  0,
  0,
  0,
  13,
  -100,
  -15,
  13,
  0,
  0,
  0,
  0,
  13,
  -100,
  -15,
  9,
  0,
  0,
  0,
  0,
  11,
  -100,
  -15,
  1,
  0,
  0,
  0,
  0,
  5,
  -100,
  -12,
  7,
  5,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  3,
  9,
  -100,
  -11,
  13,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  7,
  -100,
  -101
];
var lt17 = [
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -3,
  13,
  5,
  0,
  3,
  -5,
  9,
  3,
  1,
  0,
  0,
  3,
  -100,
  -1,
  11,
  3,
  0,
  0,
  0,
  0,
  -3,
  7,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  -100,
  9,
  0,
  0,
  0,
  0,
  0,
  0,
  -2,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  -100,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  11,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  -100,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  -1,
  7,
  0,
  0,
  0,
  0,
  0,
  0,
  7,
  13,
  -2,
  11,
  7,
  3,
  1,
  3,
  11,
  -100,
  -2,
  3,
  0,
  0,
  0,
  0,
  3,
  -100,
  -2,
  7,
  0,
  0,
  0,
  0,
  9,
  -100,
  -2,
  11,
  0,
  0,
  0,
  0,
  13,
  -100,
  -2,
  13,
  0,
  0,
  0,
  0,
  -100,
  -3,
  0,
  0,
  0,
  0,
  -100,
  -3,
  0,
  0,
  0,
  0,
  13,
  -100,
  -3,
  0,
  0,
  0,
  0,
  13,
  -100,
  -3,
  0,
  0,
  0,
  0,
  13,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  0,
  0,
  0,
  0,
  9,
  -100,
  -1,
  13,
  3,
  0,
  0,
  0,
  0,
  1,
  13,
  -100,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  13,
  -100,
  5,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  13,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -101
];
var lt18 = [
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -3,
  13,
  7,
  1,
  0,
  1,
  3,
  9,
  11,
  9,
  9,
  13,
  -100,
  -2,
  13,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  -100,
  -2,
  1,
  0,
  5,
  13,
  -1,
  13,
  11,
  3,
  0,
  0,
  0,
  1,
  -100,
  -1,
  9,
  0,
  1,
  -6,
  9,
  0,
  0,
  0,
  13,
  -100,
  -1,
  3,
  0,
  9,
  -7,
  11,
  1,
  0,
  9,
  -100,
  -1,
  0,
  0,
  -9,
  13,
  1,
  5,
  -100,
  -1,
  0,
  0,
  11,
  -9,
  13,
  11,
  -100,
  13,
  0,
  0,
  1,
  13,
  -100,
  13,
  0,
  0,
  0,
  1,
  9,
  -100,
  -1,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  9,
  -100,
  -1,
  5,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  7,
  13,
  -100,
  -1,
  13,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  -2,
  13,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  7,
  -100,
  -4,
  11,
  5,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  7,
  -100,
  -7,
  9,
  5,
  0,
  0,
  0,
  0,
  0,
  0,
  13,
  -100,
  -9,
  13,
  7,
  0,
  0,
  0,
  0,
  7,
  -100,
  -11,
  13,
  3,
  0,
  0,
  3,
  -100,
  -13,
  0,
  0,
  0,
  -100,
  7,
  5,
  -11,
  1,
  0,
  0,
  -100,
  1,
  0,
  5,
  -10,
  1,
  0,
  3,
  -100,
  0,
  0,
  0,
  3,
  -9,
  0,
  0,
  9,
  -100,
  1,
  0,
  0,
  0,
  3,
  13,
  -6,
  5,
  0,
  3,
  -100,
  7,
  0,
  0,
  0,
  0,
  1,
  7,
  11,
  13,
  13,
  9,
  3,
  0,
  1,
  13,
  -100,
  13,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  13,
  -100,
  -1,
  5,
  3,
  9,
  7,
  3,
  1,
  0,
  0,
  1,
  5,
  11,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -101
];
var lt19 = [
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -6,
  5,
  0,
  5,
  -100,
  -5,
  13,
  0,
  0,
  0,
  7,
  -100,
  -5,
  7,
  0,
  0,
  0,
  9,
  -100,
  -5,
  3,
  0,
  0,
  0,
  13,
  -100,
  -4,
  11,
  0,
  0,
  0,
  0,
  -100,
  -4,
  3,
  0,
  0,
  0,
  0,
  -100,
  -3,
  5,
  0,
  0,
  0,
  0,
  1,
  -100,
  -2,
  3,
  0,
  0,
  0,
  0,
  0,
  7,
  -100,
  13,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  11,
  -100,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  13,
  -100,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  -100,
  -3,
  9,
  0,
  0,
  0,
  0,
  7,
  13,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  1,
  0,
  0,
  0,
  11,
  -100,
  -4,
  1,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  1,
  0,
  0,
  0,
  11,
  -100,
  -4,
  3,
  0,
  0,
  0,
  11,
  -100,
  -4,
  5,
  0,
  0,
  0,
  11,
  -100,
  -4,
  3,
  0,
  0,
  0,
  11,
  -100,
  -4,
  3,
  0,
  0,
  0,
  11,
  -100,
  -4,
  1,
  0,
  0,
  0,
  9,
  -100,
  -4,
  0,
  0,
  0,
  0,
  5,
  -100,
  -4,
  0,
  0,
  0,
  0,
  1,
  -100,
  -4,
  1,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  3,
  0,
  0,
  0,
  0,
  1,
  13,
  -1,
  9,
  5,
  1,
  3,
  -100,
  -4,
  9,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  -100,
  -5,
  5,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  -100,
  -6,
  9,
  3,
  1,
  0,
  0,
  1,
  5,
  11,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -101
];
var lt20 = [
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -4,
  13,
  9,
  1,
  0,
  13,
  -11,
  11,
  5,
  1,
  5,
  -100,
  -2,
  11,
  3,
  0,
  0,
  0,
  0,
  11,
  -8,
  11,
  5,
  0,
  0,
  0,
  0,
  1,
  -100,
  13,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  11,
  -6,
  7,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  7,
  -100,
  5,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  11,
  -5,
  5,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  11,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  11,
  -5,
  9,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  -2,
  11,
  1,
  0,
  0,
  0,
  0,
  11,
  -6,
  13,
  7,
  3,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  9,
  0,
  0,
  0,
  0,
  11,
  -9,
  9,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  11,
  0,
  0,
  0,
  0,
  11,
  -10,
  0,
  0,
  0,
  0,
  11,
  -100,
  -3,
  13,
  0,
  0,
  0,
  0,
  11,
  -10,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -10,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -10,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -10,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -10,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -10,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -10,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -10,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  0,
  0,
  0,
  0,
  11,
  -10,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  1,
  0,
  0,
  0,
  11,
  -9,
  13,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  1,
  0,
  0,
  0,
  11,
  -9,
  5,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  5,
  0,
  0,
  0,
  7,
  -8,
  7,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  7,
  0,
  0,
  0,
  0,
  13,
  -5,
  13,
  5,
  0,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  -4,
  13,
  0,
  0,
  0,
  0,
  1,
  11,
  -1,
  13,
  11,
  5,
  0,
  0,
  7,
  5,
  0,
  0,
  0,
  0,
  3,
  13,
  -100,
  -5,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  13,
  -2,
  0,
  0,
  0,
  0,
  0,
  0,
  7,
  -100,
  -5,
  11,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  9,
  -4,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  -100,
  -6,
  13,
  5,
  1,
  0,
  1,
  3,
  9,
  -6,
  3,
  1,
  3,
  7,
  9,
  11,
  13,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -101
];
var lt21 = [
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -6,
  13,
  9,
  1,
  3,
  13,
  -5,
  13,
  1,
  0,
  0,
  0,
  0,
  1,
  3,
  9,
  -100,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  -5,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  -5,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  13,
  -100,
  -1,
  9,
  3,
  0,
  0,
  0,
  0,
  0,
  5,
  13,
  -7,
  13,
  9,
  0,
  0,
  0,
  0,
  1,
  11,
  -100,
  -3,
  7,
  0,
  0,
  0,
  0,
  -11,
  7,
  0,
  0,
  5,
  -100,
  -4,
  0,
  0,
  0,
  0,
  9,
  -10,
  9,
  0,
  5,
  -100,
  -4,
  5,
  0,
  0,
  0,
  1,
  -10,
  5,
  0,
  11,
  -100,
  -4,
  11,
  0,
  0,
  0,
  0,
  9,
  -9,
  0,
  1,
  -100,
  -5,
  1,
  0,
  0,
  0,
  3,
  -8,
  9,
  0,
  7,
  -100,
  -5,
  7,
  0,
  0,
  0,
  0,
  13,
  -7,
  3,
  0,
  13,
  -100,
  -5,
  13,
  0,
  0,
  0,
  0,
  7,
  -6,
  13,
  0,
  3,
  -100,
  -6,
  3,
  0,
  0,
  0,
  1,
  -6,
  7,
  0,
  11,
  -100,
  -6,
  9,
  0,
  0,
  0,
  0,
  9,
  -5,
  0,
  1,
  -100,
  -6,
  13,
  0,
  0,
  0,
  0,
  3,
  -4,
  7,
  0,
  7,
  -100,
  -7,
  5,
  0,
  0,
  0,
  0,
  11,
  -3,
  1,
  0,
  13,
  -100,
  -7,
  11,
  0,
  0,
  0,
  0,
  1,
  -2,
  9,
  0,
  5,
  -100,
  -8,
  1,
  0,
  0,
  0,
  0,
  5,
  -1,
  1,
  0,
  11,
  -100,
  -8,
  7,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  -100,
  -8,
  13,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  -9,
  3,
  0,
  0,
  0,
  0,
  0,
  1,
  -100,
  -9,
  9,
  0,
  0,
  0,
  0,
  0,
  7,
  -100,
  -10,
  0,
  0,
  0,
  0,
  0,
  13,
  -100,
  -10,
  5,
  0,
  0,
  0,
  1,
  -100,
  -10,
  13,
  1,
  0,
  0,
  7,
  -100,
  -11,
  11,
  1,
  5,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -101
];
var lt22 = [
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -29,
  13,
  -100,
  5,
  0,
  0,
  1,
  1,
  1,
  3,
  3,
  5,
  7,
  -3,
  7,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  11,
  -4,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  3,
  11,
  -100,
  5,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  7,
  -3,
  9,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  11,
  -4,
  9,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  13,
  -100,
  -1,
  13,
  5,
  0,
  0,
  0,
  0,
  0,
  13,
  -5,
  13,
  5,
  0,
  0,
  0,
  0,
  0,
  3,
  -8,
  9,
  0,
  0,
  0,
  0,
  5,
  -100,
  -3,
  5,
  0,
  0,
  0,
  0,
  -8,
  11,
  0,
  0,
  0,
  0,
  9,
  -9,
  9,
  0,
  0,
  9,
  -100,
  -4,
  1,
  0,
  0,
  0,
  13,
  -8,
  11,
  0,
  0,
  0,
  3,
  -10,
  0,
  3,
  -100,
  -4,
  7,
  0,
  0,
  0,
  11,
  -8,
  13,
  0,
  0,
  0,
  0,
  13,
  -8,
  9,
  0,
  11,
  -100,
  -4,
  13,
  0,
  0,
  0,
  7,
  -8,
  11,
  0,
  0,
  0,
  0,
  7,
  -8,
  1,
  1,
  -100,
  -5,
  5,
  0,
  0,
  1,
  -8,
  7,
  0,
  0,
  0,
  0,
  3,
  -8,
  5,
  11,
  -100,
  -5,
  11,
  0,
  0,
  0,
  9,
  -7,
  1,
  0,
  0,
  0,
  0,
  0,
  13,
  -100,
  -6,
  0,
  0,
  0,
  3,
  -6,
  9,
  0,
  3,
  13,
  0,
  0,
  0,
  7,
  -6,
  5,
  -100,
  -6,
  3,
  0,
  0,
  0,
  13,
  -5,
  1,
  0,
  11,
  -1,
  3,
  0,
  0,
  3,
  -6,
  1,
  -100,
  -6,
  9,
  0,
  0,
  0,
  9,
  -4,
  11,
  0,
  3,
  -2,
  7,
  0,
  0,
  0,
  13,
  -4,
  11,
  1,
  -100,
  -6,
  13,
  0,
  0,
  0,
  3,
  -4,
  5,
  0,
  9,
  -2,
  13,
  0,
  0,
  0,
  7,
  -4,
  5,
  3,
  -100,
  -7,
  3,
  0,
  0,
  0,
  13,
  -3,
  1,
  1,
  -4,
  0,
  0,
  0,
  1,
  -4,
  1,
  7,
  -100,
  -7,
  9,
  0,
  0,
  0,
  5,
  -2,
  13,
  0,
  7,
  -4,
  5,
  0,
  0,
  0,
  9,
  -2,
  9,
  0,
  13,
  -100,
  -8,
  1,
  0,
  0,
  0,
  13,
  -1,
  7,
  1,
  -5,
  11,
  0,
  0,
  0,
  1,
  -2,
  1,
  3,
  -100,
  -8,
  7,
  0,
  0,
  0,
  1,
  13,
  1,
  5,
  -6,
  1,
  0,
  0,
  0,
  13,
  9,
  0,
  9,
  -100,
  -8,
  13,
  0,
  0,
  0,
  0,
  0,
  0,
  11,
  -6,
  7,
  0,
  0,
  0,
  1,
  0,
  1,
  -100,
  -9,
  3,
  0,
  0,
  0,
  0,
  3,
  -7,
  13,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  -9,
  9,
  0,
  0,
  0,
  0,
  5,
  -8,
  3,
  0,
  0,
  0,
  0,
  13,
  -100,
  -9,
  13,
  0,
  0,
  0,
  0,
  11,
  -8,
  9,
  0,
  0,
  0,
  1,
  -100,
  -10,
  3,
  0,
  0,
  0,
  -10,
  3,
  0,
  0,
  5,
  -100,
  -10,
  11,
  0,
  0,
  7,
  -10,
  13,
  1,
  1,
  13,
  -100,
  -11,
  11,
  11,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -101
];
var lt23 = [
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -16,
  13,
  13,
  -1,
  13,
  9,
  5,
  1,
  0,
  1,
  5,
  13,
  -100,
  -1,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  7,
  -4,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  -100,
  -1,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  11,
  -4,
  7,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  -100,
  -2,
  13,
  7,
  0,
  0,
  0,
  0,
  0,
  0,
  13,
  -6,
  13,
  5,
  0,
  0,
  0,
  0,
  5,
  9,
  13,
  -100,
  -4,
  13,
  3,
  0,
  0,
  0,
  0,
  3,
  -7,
  13,
  0,
  0,
  5,
  13,
  -100,
  -6,
  3,
  0,
  0,
  0,
  0,
  5,
  -6,
  7,
  0,
  7,
  -100,
  -7,
  3,
  0,
  0,
  0,
  0,
  11,
  -4,
  11,
  0,
  9,
  -100,
  -8,
  5,
  0,
  0,
  0,
  1,
  -3,
  11,
  0,
  7,
  -100,
  -9,
  3,
  0,
  0,
  0,
  1,
  11,
  11,
  0,
  7,
  -100,
  -9,
  13,
  1,
  0,
  0,
  0,
  0,
  0,
  7,
  -100,
  -10,
  13,
  1,
  0,
  0,
  0,
  0,
  9,
  -100,
  -11,
  11,
  0,
  0,
  0,
  0,
  1,
  -100,
  -12,
  0,
  0,
  0,
  0,
  0,
  7,
  -100,
  -11,
  9,
  0,
  0,
  0,
  0,
  0,
  0,
  11,
  -100,
  -10,
  9,
  0,
  5,
  13,
  3,
  0,
  0,
  0,
  1,
  13,
  -100,
  -9,
  11,
  0,
  3,
  -2,
  13,
  1,
  0,
  0,
  0,
  1,
  13,
  -100,
  -8,
  11,
  0,
  3,
  13,
  -3,
  13,
  1,
  0,
  0,
  0,
  1,
  13,
  -100,
  -7,
  9,
  0,
  1,
  13,
  -5,
  11,
  0,
  0,
  0,
  0,
  1,
  13,
  -100,
  -6,
  9,
  0,
  0,
  7,
  -7,
  9,
  0,
  0,
  0,
  0,
  1,
  11,
  -100,
  -5,
  5,
  0,
  0,
  0,
  13,
  -8,
  7,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  -3,
  13,
  3,
  0,
  0,
  0,
  0,
  13,
  -8,
  13,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  -100,
  9,
  3,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  11,
  -6,
  11,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  3,
  13,
  -100,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  -6,
  11,
  5,
  3,
  1,
  0,
  0,
  0,
  1,
  1,
  3,
  5,
  -100,
  13,
  13,
  13,
  13,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -101
];
var lt24 = [
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -100,
  -1,
  13,
  13,
  -2,
  13,
  13,
  11,
  11,
  11,
  13,
  -6,
  11,
  11,
  11,
  11,
  11,
  11,
  11,
  11,
  13,
  -100,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  -4,
  1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  -100,
  7,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  5,
  -4,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  -2,
  11,
  5,
  0,
  0,
  0,
  0,
  0,
  0,
  11,
  -6,
  11,
  0,
  0,
  0,
  0,
  1,
  7,
  13,
  -100,
  -4,
  11,
  0,
  0,
  0,
  0,
  0,
  -8,
  9,
  0,
  0,
  1,
  13,
  -100,
  -5,
  7,
  0,
  0,
  0,
  0,
  13,
  -7,
  9,
  0,
  0,
  13,
  -100,
  -6,
  1,
  0,
  0,
  0,
  9,
  -7,
  5,
  0,
  5,
  -100,
  -6,
  7,
  0,
  0,
  0,
  3,
  -7,
  1,
  0,
  13,
  -100,
  -6,
  13,
  0,
  0,
  0,
  0,
  13,
  -5,
  11,
  0,
  1,
  -100,
  -7,
  5,
  0,
  0,
  0,
  7,
  -5,
  7,
  0,
  7,
  -100,
  -7,
  11,
  0,
  0,
  0,
  3,
  -5,
  1,
  0,
  11,
  -100,
  -8,
  1,
  0,
  0,
  0,
  13,
  -3,
  13,
  0,
  3,
  -100,
  -8,
  7,
  0,
  0,
  0,
  7,
  -3,
  7,
  0,
  9,
  -100,
  -8,
  13,
  0,
  0,
  0,
  3,
  -3,
  1,
  1,
  -100,
  -9,
  5,
  0,
  0,
  0,
  13,
  -1,
  9,
  0,
  7,
  -100,
  -9,
  11,
  0,
  0,
  0,
  3,
  13,
  1,
  0,
  13,
  -100,
  -10,
  1,
  0,
  0,
  0,
  0,
  0,
  3,
  -100,
  -10,
  9,
  0,
  0,
  0,
  0,
  0,
  9,
  -100,
  -11,
  1,
  0,
  0,
  0,
  1,
  -100,
  -11,
  5,
  0,
  0,
  0,
  7,
  -100,
  -11,
  9,
  0,
  0,
  0,
  13,
  -100,
  -11,
  13,
  0,
  0,
  3,
  -100,
  -12,
  0,
  0,
  9,
  -100,
  -12,
  0,
  1,
  -100,
  -11,
  13,
  0,
  7,
  -100,
  -11,
  7,
  0,
  11,
  -100,
  -11,
  1,
  0,
  -100,
  -2,
  7,
  0,
  1,
  7,
  -4,
  11,
  0,
  3,
  -100,
  -1,
  11,
  0,
  0,
  0,
  0,
  1,
  9,
  13,
  11,
  1,
  0,
  9,
  -100,
  -1,
  3,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  3,
  -100,
  -1,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  13,
  -100,
  -1,
  7,
  0,
  0,
  0,
  0,
  0,
  0,
  0,
  1,
  13,
  -100,
  -2,
  9,
  3,
  0,
  0,
  1,
  3,
  7,
  -100,
  -101
];
var allLetters = [
  lt0,
  lt1,
  lt2,
  lt3,
  lt4,
  lt5,
  lt6,
  lt7,
  lt8,
  lt9,
  lt10,
  lt11,
  lt12,
  lt13,
  lt14,
  lt15,
  lt16,
  lt17,
  lt18,
  lt19,
  lt20,
  lt21,
  lt22,
  lt23,
  lt24
];

// src/lib/Sw.ts
var SW = [
  0,
  4,
  8,
  12,
  16,
  20,
  23,
  27,
  31,
  35,
  39,
  43,
  47,
  50,
  54,
  58,
  61,
  65,
  68,
  71,
  75,
  78,
  81,
  84,
  87,
  90,
  93,
  96,
  98,
  101,
  103,
  105,
  108,
  110,
  112,
  114,
  115,
  117,
  119,
  120,
  121,
  122,
  123,
  124,
  125,
  126,
  126,
  127,
  127,
  127,
  127,
  127,
  127,
  127,
  126,
  126,
  125,
  124,
  123,
  122,
  121,
  120,
  119,
  117,
  115,
  114,
  112,
  110,
  108,
  105,
  103,
  101,
  98,
  96,
  93,
  90,
  87,
  84,
  81,
  78,
  75,
  71,
  68,
  65,
  61,
  58,
  54,
  50,
  47,
  43,
  39,
  35,
  31,
  27,
  23,
  20,
  16,
  12,
  8,
  4,
  0,
  -4,
  -8,
  -12,
  -16,
  -20,
  -23,
  -27,
  -31,
  -35,
  -39,
  -43,
  -47,
  -50,
  -54,
  -58,
  -61,
  -65,
  -68,
  -71,
  -75,
  -78,
  -81,
  -84,
  -87,
  -90,
  -93,
  -96,
  -98,
  -101,
  -103,
  -105,
  -108,
  -110,
  -112,
  -114,
  -115,
  -117,
  -119,
  -120,
  -121,
  -122,
  -123,
  -124,
  -125,
  -126,
  -126,
  -127,
  -127,
  -127,
  -127,
  -127,
  -127,
  -127,
  -126,
  -126,
  -125,
  -124,
  -123,
  -122,
  -121,
  -120,
  -119,
  -117,
  -115,
  -114,
  -112,
  -110,
  -108,
  -105,
  -103,
  -101,
  -98,
  -96,
  -93,
  -90,
  -87,
  -84,
  -81,
  -78,
  -75,
  -71,
  -68,
  -65,
  -61,
  -58,
  -54,
  -50,
  -47,
  -43,
  -39,
  -35,
  -31,
  -27,
  -23,
  -20,
  -16,
  -12,
  -8,
  -4
];

// src/lib/Captcha.ts
var Captcha = class {
  numberOfDots;
  gifSize = 17646;
  letters = "abcdefghijklmnopqrstuvwxyz";
  blurImg;
  filterImg;
  constructor({ numberOfDots = 100, blur = false, filter = false } = {}) {
    this.numberOfDots = numberOfDots;
    this.blurImg = blur;
    this.filterImg = filter;
  }
  /**
   *
   * @param size The number of letters
   * @param color The color of the text
   * @returns The gif buffer and the token
   */
  generate(size = 5, color) {
    const buffer = Buffer.alloc(this.gifSize);
    const { background, token } = this.makeCaptcha(size);
    this.makeGif(background, buffer, color);
    return {
      buffer,
      token
    };
  }
  /**
   *
   * @param min min size
   * @param max max size
   * @returns A number between min and max
   */
  random(min, max) {
    return Math.floor(Math.random() * (max - min)) + min;
  }
  /**
   *
   * @param n The index of the letter
   * @param pos The position of the letter
   * @param background The background buffer
   * @param swr The sw buffer
   * @param s1 The s1 value
   * @param s2 The s2 value
   * @returns The new position of the letter
   */
  letter(n, pos, background, swr, s1, s2) {
    const l = background.length;
    const t = allLetters[n];
    let r = 200 * 16 + pos;
    let i = r;
    let sk1 = s1 + pos;
    let sk2 = s2 + pos;
    let mpos = pos;
    let row = 0;
    for (let j = 0, k = t.length; j < k; j++) {
      const p = t[j];
      if (p === -101)
        continue;
      if (p < 0) {
        if (p === -100) {
          r += 200;
          i = r;
          sk1 = s1 + pos;
          row++;
          continue;
        }
        i += -p;
        continue;
      }
      if (sk1 >= 200)
        sk1 %= 200;
      const skew = Math.floor(SW[sk1] / 16);
      sk1 += (swr[pos + i - r] & 1) + 1;
      if (sk2 >= 200)
        sk2 %= 200;
      const skewh = Math.floor(SW[sk2] / 70);
      sk2 += swr[row] & 1;
      const x = i + skew * 200 + skewh;
      mpos = Math.max(mpos, pos + i - r);
      if (x - l < 70 * 200)
        background[x] = p << 4;
      i++;
    }
    return mpos;
  }
  /**
   *
   * @param background The background buffer
   * @param swr The sw buffer
   * @param s1 The s1 value
   */
  line(background, swr, s1) {
    for (let x = 0, sk1 = s1; x < 199; x++) {
      if (sk1 >= 200)
        sk1 %= 200;
      const skew = Math.floor(SW[sk1] / 16);
      sk1 += (swr[x] & 3) + 1;
      const i = 200 * (45 + skew) + x;
      background[i + 0] = 0;
      background[i + 1] = 0;
      background[i + 200] = 0;
      background[i + 201] = 0;
    }
  }
  /**
   *
   * @param background The background buffer
   * @param dr The dr buffer
   */
  dots(background, dr) {
    for (let n = 0; n < this.numberOfDots; n++) {
      const v = dr.readUInt32BE(n);
      const i = v % (200 * 67);
      background[i + 0] = 255;
      background[i + 1] = 255;
      background[i + 2] = 255;
      background[i + 200] = 255;
      background[i + 201] = 255;
      background[i + 202] = 255;
    }
  }
  /**
   *
   * @param background The background buffer
   */
  blur(background) {
    for (let i = 0, y = 0; y < 68; y++) {
      for (let x = 0; x < 198; x++) {
        const c11 = background[i + 0];
        const c12 = background[i + 1];
        const c21 = background[i + 200];
        const c22 = background[i + 201];
        background[i++] = Math.floor((c11 + c12 + c21 + c22) / 4);
      }
    }
  }
  /**
   *
   * @param background The background buffer
   */
  filter(background) {
    const om = Buffer.alloc(70 * 200).fill(255);
    let i = 0;
    let o = 0;
    for (let y = 0; y < 70; y++) {
      for (let x = 4; x < 200 - 4; x++) {
        if (background[i] > 240 && background[i + 1] < 240) {
          om[o] = 0;
          om[o + 1] = 0;
        } else if (background[i] < 240 && background[i + 1] > 240) {
          om[o] = 0;
          om[o + 1] = 0;
        }
        i++;
        o++;
      }
    }
    om.copy(background);
  }
  /**
   *
   * @param size The number of letters
   * @returns The background buffer and the token
   */
  makeCaptcha(size) {
    const rb = crypto.randomBytes(size + 200 + 100 * 4 + 1 + 1);
    const tb = Buffer.alloc(size);
    const swr = Buffer.alloc(200);
    const dr = Buffer.alloc(100 * 4);
    rb.copy(tb, 0, 0, size);
    rb.copy(swr, 0, size, size + 200);
    rb.copy(dr, 0, size + 200, size + 200 + 100 * 4);
    let s1 = rb.readUInt8(size + 200 + 100 * 4);
    let s2 = rb.readUInt8(size + 200 + 100 * 4 + 1);
    const background = Buffer.alloc(200 * 70).fill(255);
    s1 &= 127;
    s2 &= 63;
    let p = 30;
    for (let n = 0; n < size; n++) {
      tb[n] %= 25;
      p = this.letter(tb[n], p, background, swr, s1, s2);
      tb[n] = this.letters.charCodeAt(tb[n]);
    }
    this.line(background, swr, s1);
    this.dots(background, dr);
    if (this.blurImg)
      this.blur(background);
    if (this.filterImg)
      this.filter(background);
    return { background, token: tb.toString() };
  }
  /**
   *
   * @param background The background buffer
   * @param gif The gif buffer
   * @param color The color of the text
   *
   */
  makeGif(background, gif, color) {
    const r = color ?? this.random(0, colors.length);
    gif.fill(colors[r].replace(/\n/g, ""), 0, 13 + 48 + 10 + 1, "ascii");
    let i = 0;
    let p = 13 + 48 + 10 + 1;
    for (let y = 0; y < 70; y++) {
      gif[p++] = 250;
      for (let x = 0; x < 50; x++) {
        const a = background[i + 0] >> 4;
        const b = background[i + 1] >> 4;
        const c = background[i + 2] >> 4;
        const d = background[i + 3] >> 4;
        gif[p + 0] = 16 | a << 5;
        gif[p + 1] = a >> 3 | 64 | b << 7;
        gif[p + 2] = b >> 1;
        gif[p + 3] = 1 | c << 1;
        gif[p + 4] = 4 | d << 3;
        i += 4;
        p += 5;
      }
    }
    gif.fill("\0;", this.gifSize - 4);
  }
};
__name(Captcha, "Captcha");

exports.Captcha = Captcha;
